app/models/content_item.rb

```diff
@@ -152,9 +163,17 @@ def includes_auth_bypass_id?(auth_bypass_id)
   end
 
   def register_routes(previous_item: nil)
-    return if self.schema_name.start_with?("placeholder")
-    return if previous_item && previous_item.route_set == self.route_set
-    self.route_set.register!
+    return unless should_register_routes?(previous_item: previous_item)
+
+    3.times do
```

Comment:
I'd be inclined to make this number a constant or put it in config if only to give it a label.